tests/MongoDB.Driver.Tests/AggregateFluentTests.cs

```diff
@@ -1031,6 +1031,80 @@ public void ReplaceWith_should_add_the_expected_stage([Values(false, true)] bool
             }
         }
 
+        public class Item1
+        {
+            public ObjectId Id { get; set; }
+            public string Name1 { get; set; }
+        }
+
+        public class Item2
+        {
+            public ObjectId Id { get; set; }
+            public string Name2 { get; set; }
+        }
+
+        [SkippableFact]
+        [ParameterAttributeData]
+        public void UnionWith_with_different_schemas_and_projection_should_return_the_expected_result()
+        {
+            RequireServer.Check().Supports(Feature.AggregateUnionWith);
+
+            var databaseName = "test";
+            var items1CollectionName = "items1";
+            var items2CollectionName = "items2";
+
+            var client = CreateClient();
+            DropCollection(client, databaseName, items1CollectionName);
+            DropCollection(client, databaseName, items2CollectionName);
+
+            var db = client.GetDatabase(databaseName);
+            var items1Collection = db.GetCollection<Item1>(items1CollectionName);
+            var items2Collection = db.GetCollection<Item2>(items2CollectionName);
+
+            var item1Documents = new[]
+            {
+                new Item1 { Name1 = "almonds" },
+                new Item1 { Name1 = "cookies" }
+            };
+            items1Collection.InsertMany(item1Documents);
+
+            var item2Documents = new[]
+            {
+                new Item2 { Name2 = "cookies" },
+                new Item2 { Name2 = "cookies" },
+                new Item2 { Name2 = "pecans" }
+            };
+            items2Collection.InsertMany(item2Documents);
+
+            var withPipeline =
+                new EmptyPipelineDefinition<Item2>()
+                .Match(x => x.Name2 == "cookies")
+                .Project(
+                    item2 => 
+                        new Item1
+                        {
+                            Name1 = item2.Name2 // manually project Item2 into Item1
+                        });
+
+            var result = items1Collection
+                .Aggregate()
+                .UnionWith<Item2>(items2Collection, withPipeline)
+                .ToList()
+                .Select(item =>
+                {
+                    var document = item.ToBsonDocument();
+                    document.Remove("_id");
+                    return document;
+                })
+                .ToList();
+
+            result.Count.Should().Be(4);
+            result[0].Should().Be("{ Name1 : 'almonds' }");
+            result[1].Should().Be("{ Name1 : 'cookies' }");
+            result[2].Should().Be("{ Name1 : 'cookies' }");
+            result[3].Should().Be("{ Name1 : 'cookies' }");
```

Comment:
Lines 1102-1105 can now assert directly against the instances of `ItemResult`:

    result[0].Name.Should().Be("almonds");
    result[1].Name.Should().Be("cookies");
    result[2].Name.Should().Be("cookies");
    result[3].Name.Should().Be("cookies");